test/unit/sql/where.test.js

```diff
@@ -908,6 +908,24 @@ suite(Support.getTestDialectTeaser('SQL'), () => {
       });
     }
 
+    if (current.dialect.supports.REGEXP) {
+      suite('$regexp', () => {
+        testsql('username', {
+          $regexp: '^sw.*r$'
```

Comment:
I had the idea to support JavaScript RegExp literals like `$regexp: /^[h|a|t]$/i`, but thinking more about it, JavaScript regexps are based on Perl, which is very different from Posix, so the set of escape sequences differs, which could cause unwanted behaviour.